defender/emitVotes.js

```diff
@@ -0,0 +1,241 @@
+            continue;
+        }
+
+        if (state == 1) {
+            console.log(
+                `Proposal ${proposalId} is in the cross chain vote collection state`,
+            );
+
+            const proposalInformation =
+                await voteCollection.proposalInformation(proposalId);
+
+            const {notificationClient} = context;
+            let subject;
+            let message;
+
+            // only emit votes if proposal total votes is greater than 0
+            if (proposalInformation[5] > 0) {
+                try {
+                    const bridgeCost = await voteCollection.bridgeCostAll();
+
+                    const gasEstimate =
+                        await voteCollection.estimateGas.emitVotes(proposalId, {
+                            value: bridgeCost,
+                        });
+
+                    // Add a buffer to the gas estimate
+                    const gasLimit = gasEstimate.add(gasEstimate.div(5)); // Adding 20% extra gas as a buffer
+                    await voteCollection.emitVotes(proposalId, {
+                        value: bridgeCost,
+                        gasLimit: gasLimit,
+                    });
+
+                    subject = `Votes emitted for proposal ${proposalId} on ${network}`;
+                    message = `If the proposal reaches quorum, it will be executed automatically once the cross-chain vote collection period finishes.`;
+                } catch (error) {
+                    console.error(
+                        `Error emitting votes for proposal ${proposalId}: ${error.message}`,
+                    );
+                    subject = `Error emitting votes for proposal ${proposalId} on ${network}`;
+                    message = `Error: ${error.message}. Manual intervention required.`;
+                }
+
+                if (kvStoreValue) {
+                    kvStoreValue += `,${proposalId}`;
+                } else {
+                    kvStoreValue = `${proposalId}`;
+                }
+
+                console.log(`Storing proposal ${proposalId} in KV store...`);
+                await kvStore.put(`${network}`, kvStoreValue);
+                console.log(`Updated KV store value for ${network}`);
+
+                // Slack
+                notificationClient.send({
+                    channelAlias: 'Parameter Changes to Slack',
+                    subject,
+                    message,
+                });
+
+                // Discord
+                const {GOVBOT_WEBHOOK} = event.secrets;
+                const moonwellEvent = new MoonwellEvent();
+                const discordPayload = moonwellEvent.discordMessagePayload(
+                    0x00ff00,
+                    subject,
+                    message,
+                );
+                await moonwellEvent.sendDiscordMessage(
+                    GOVBOT_WEBHOOK,
+                    discordPayload,
+                );
+
+                return;
```

Comment:
I think this return should be removed as it will break the for loop for further proposals.
```suggestion
```